newsroom/reports/reports.py

```diff
@@ -260,37 +264,98 @@ async def get_subscriber_activity_report():
         if doc.get("company"):
             company_ids.append(ObjectId(doc.get("company")))
         user_ids.append(ObjectId(doc.get("user")))
-    agenda_items = get_entity_dict(get_items_by_id(agenda_ids, "agenda"))
-    wire_items = get_entity_dict(get_items_by_id(wire_ids, "items"))
-    company_items = get_entity_dict(get_items_by_id(company_ids, "companies"), True)
-    user_items = get_entity_dict(get_items_by_id(user_ids, "users"), True)
+    # remove duplicates for efficiency
+    wire_ids = list(set(wire_ids))
+    agenda_ids = list(set(agenda_ids))
+
+    AGENDAITEM_CHUNK_SIZE: int = 100
+    agenda_items: dict[str, AgendaItem] = {}
+    for i in range(0, len(wire_ids), AGENDAITEM_CHUNK_SIZE):
+        agenda_cursor: ResourceCursorAsync[AgendaItem] = await AgendaItemService().search(
+            {"_id": {"$in": agenda_ids[i : i + AGENDAITEM_CHUNK_SIZE]}}, use_mongo=True
+        )
+        agenda_items: dict[str, AgendaItem] = {
```

Comment:
This is overwriting the `agenda_items` variable, should this rather be updating it instead?